app/src/main/java/com/andrew/tasky/agenda/presentation/screens/agenda/AgendaViewModel.kt

```diff
@@ -134,4 +136,17 @@ class AgendaViewModel@Inject constructor(
             }
         }
     }
+
+    private val syncModifiedAgendaItemsWorkRequest =
+        PeriodicWorkRequestBuilder<SyncModifiedAgendaItemsWorker>(15, TimeUnit.MINUTES)
+            .setConstraints(
+                Constraints.Builder()
+                    .setRequiredNetworkType(NetworkType.CONNECTED)
+                    .build()
+            ).build()
+
+    init {
+        viewModelScope.launch { updateAgendaOfDate(date = dateSelected.value) }
+        workManager.enqueue(syncModifiedAgendaItemsWorkRequest)
+    }
```

Comment:
Using WorkManager directly in a ViewModel prevents that it can be locally unit tested since WorkManager is an Android dependency. You could do this to solve this:
```
// In domain
interface AgendaItemScheduler {
    fun scheduleSync(minuteInterval: Int)
}

// In data
class AgendaItemSchedulerImpl(val workManager: WorkManager): AgendaItemScheduler {
    // Implement function
}

// In ViewModel
class AgendaViewModel(val scheduler: AgendaItemScheduler): ViewModel() {
    init {
        scheduler.scheduleSync(minuteInterval = 15)
    }
}
```

This way, you could pass another implementation of `AgendaItemScheduler` to your ViewModel in a test case which does not depend on WorkManager